File: Algorithms/Sorting/OddEvenSort.cpp
#include<bits/stdc++.h>
using namespace std;

void swap(int *, int *);
void oddevenSort(int *, int );

int main()
{
    int a[1000], i, n;
    printf("Enter the number of elements: ");
    scanf("%d", &n);

    printf("Enter elements:\n");
    for (i=0; i < n; i++)
        scanf("%d", &a[i]);

    oddevenSort(a, n);

    printf("Sorted elements are:\n");
    for (i = 0; i < n; i++)
        printf("%d ", a[i]);

    return 0;
}

void swap(int * x, int * y) {
    int temp = *x;
    *x = *y;
    *y = temp;
}

void oddevenSort(int *x, int n) {
    int sort = 0, i;
    while (!sort) {
        sort = 1;
        for (i=1; i<n-1; i += 2) {
            if (x[i] > x[i+1]) {
                swap(&x[i], &x[i+1]);
                sort = 0;
            }
        }

        for (i = 0; i<n; i += 2) {
            if (x[i] > x[i + 1]) {
                swap(&x[i], &x[i + 1]);
                sort = 0;
            }
        }
    }
}

File: Algorithms/Sorting/OddEvenSort.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text.RegularExpressions;

namespace Rextester
{
    public static class Program
    {
        public static void Main(string[] args)
        {
            new[] {7, 9, 8, 7, 3, 15, 6, 6, 4, 3}.Sort().Flatten().Output();
        }

        static int[] Sort(this int[] array)
        {
            var sorted = false;
            while(!sorted)
            {
                sorted = true;
                for(var i = 1; i < array.Length-1; i += 2)
                {
                    if(array[i] > array[i+1])
                    {
                        Swap(array, i, i+1);
                        sorted = false;
                    }
                }
 
                for(var i = 0; i < array.Length-1; i += 2)
                {
                    if(array[i] > array[i+1])
                    {
                        Swap(array, i, i+1);
                        sorted = false;
                    }
                }
            }
            return array;
        }

        static void Swap(int[] array, int i, int j)
        {
            var temp = array[i];
            array[i] = array[j];
            array[j] = temp;
        }

        static string Flatten(this int[] array)
        {
            return array.Select(x => x.ToString()).Aggregate((acc, x) => acc + ";" + x);
        }

        static void Output(this string output)
        {
            Console.WriteLine(output);
        }
    }
}